pkg/bisect/bisect.go

```diff
@@ -460,25 +464,51 @@ func (env *env) test() (*testResult, error) {
 		}
 		return res, nil
 	}
-	testStart := time.Now()
-	results, err := env.inst.Test(NumTests, cfg.Repro.Syz, cfg.Repro.Opts, cfg.Repro.C)
-	env.testTime += time.Since(testStart)
-	if err != nil {
-		env.log("failed: %v", err)
-		return res, nil
-	}
-	bad, good, rep := env.processResults(current, results)
-	res.rep = rep
-	res.verdict = vcs.BisectSkip
-	if bad != 0 {
-		res.verdict = vcs.BisectBad
-	} else if NumTests-good-bad > NumTests/3*2 {
-		// More than 2/3 of instances failed with infrastructure error,
-		// can't reliably tell that the commit is good.
-		res.verdict = vcs.BisectSkip
-	} else if good != 0 {
-		res.verdict = vcs.BisectGood
+
+	// Increase number of tests in case reproducer is flaky.
+	flakyNumTests := 0
+	if env.flaky {
+		flakyNumTests = NumTests
```

Comment:
This is not this simple, see:
https://github.com/google/syzkaller/blob/4a7fa9b416fd0a961793328a785666d6d2c5976d/syz-ci/jobs.go#L363-L367
Consider you are testing with 3 android phones with their IDs listed in the config, we can't suddenly materialize more of the phone out of thin air :)
We can do this, but only for some VM types, which is done with that hack.

I think we need `const MaxNumTests = 20` and then use `MaxNumTests/2` if test is not flaky.

Reply:
Ah, thank you for pointing this out. I was too much oriented testing with syz-bisect and missed this hook in jobs.go.  I will change it as suggested.